src/runtime/metal/metal_module.mm

```diff
@@ -192,8 +195,12 @@ void operator()(TVMArgs args, TVMRetValue* rv, const ArgUnion64* pack_args) cons
       int blockSize = wl.block_dim(0) * wl.block_dim(1) * wl.block_dim(2);
       auto maxTotalThreadsPerThreadgroup = scache_[device_id].maxTotalThreadsPerThreadgroup;
       CHECK_LE(blockSize, maxTotalThreadsPerThreadgroup);
-      id<MTLCommandQueue> queue = w_->GetCommandQueue(t->device);
-      id<MTLCommandBuffer> cb = [queue commandBuffer];
+      id<MTLCommandBuffer> cb = [queue->queue_ commandBuffer];
+      [cb addCompletedHandler:^(id<MTLCommandBuffer> buffer) {
```

Comment:
A lot of CompletedHandler blocks specification. It can be done inside "Queue->getCommandBuffer"